google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -353,11 +383,71 @@ private void refreshOpenTelemetryTableNameAttributes() {
     }
   }
 
+  private Attributes augmentAttributesWithErrorCode(Attributes attributes, String errorCode) {
+    AttributesBuilder builder = attributes.toBuilder();
+    if ((errorCode != null) && !errorCode.isEmpty()) {
+      builder.put(telemetryKeyErrorCode, errorCode);
+    }
+    return builder.build();
+  }
+
   @VisibleForTesting
   Attributes getTelemetryAttributes() {
     return telemetryAttributes;
   }
 
+  private void periodicallyReportOpenTelemetryMetrics() {
+    Duration durationSinceLastRefresh = Duration.between(instantLastSentMetrics, Instant.now());
+    if (durationSinceLastRefresh.compareTo(METRICS_UPDATE_INTERVAL) > 0) {
```

Comment:
Are metrics updates really that costly on the producer side that you don't just update metrics at time of event?

In opencensus, flushing/updates were mostly an exporter concern.

Reply:
I am testing using an exporter to Google Cloud Monitoring. I encountered "exceeded max frequency" errors with this exporter. To resolve this issue, I have switched to updating the instruments only once every second, which I believe should be sufficient for our needs.

Reply:
Upon further inspection, I narrowed down the issue I was seeing to the frequency of the exporter. I have restored all metrics to be instrumented in real time.